### PR TITLE
feat(kuma-cp) dataplane certificate rotation

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
### Summary

Introduce certificate rotation after 4/5 life of the certificate.
Also changed the default expiration to 30 days.

### CODE REVIEW

**Comment by subnetmarco** [comment]
Does this PR also introduce new properties for `mtls`? Also please link to the documentation PR once it's ready.

**Comment by jakubdyszkiewicz** [comment]
@subnetmarco I still need to do builtin CA expiration and rsaBits. I'll update docs after this change

File: pkg/sds/server/reconciler.go [hunk 1]
```diff
@@ -76,22 +80,37 @@ func (d *DataplaneReconciler) Reconcile(dataplaneId core_model.ResourceKey) erro
...
-		return true, nil
+		return true, fmt.Sprintf("Enabled CA changed from %s to %s", caName, mesh.GetEnabledCertificateAuthorityBackend().Name), nil
+	}
+	// generate snapshot if cert expired
+	generationUnixNano, err := strconv.Atoi(parts[0])
+	if err != nil {
+		return false, "", errors.Wrap(err, `invalid snapshot version format. Format should be "UnixNano;NameOfTheCA"`)
+	}
+	expiration := issuer.DefaultWorkloadCertValidityPeriod
+	if mesh.GetEnabledCertificateAuthorityBackend().GetDpCert().GetRotation().GetExpiration() != nil {
```
Comment (lobkovilya) [inline pkg/sds/server/reconciler.go:104]: I think makes sense to specify `MinWorkloadCertValidityPeriod` to avoid a mess with small values

Comment (jakubdyszkiewicz) [reply pkg/sds/server/reconciler.go:104]: Can you elaborate here? What will be the mess with small values?

**Pull request merged**